var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -683,16 +683,17 @@ def cmake_args(self):
                 # hence the test to see if the version starts with "flang".
                 targets.append('CppBackend')
 
-            if 'x86' in spec.architecture.target.lower():
+            target_str = str(spec.architecture.target).lower()
+            if 'x86' in target_str:
                 targets.append('X86')
-            elif 'arm' in spec.architecture.target.lower():
+            elif 'arm' in target_str:
                 targets.append('ARM')
-            elif 'aarch64' in spec.architecture.target.lower():
+            elif 'aarch64' in target_str:
                 targets.append('AArch64')
-            elif 'sparc' in spec.architecture.target.lower():
+            elif 'sparc' in target_str:
                 targets.append('Sparc')
-            elif ('ppc' in spec.architecture.target.lower() or
-                  'power' in spec.architecture.target.lower()):
+            elif ('ppc' in target_str or
+                  'power' in target_str):
```

Comment:
These should likely check against `spec.target.family`, as the `target` may now be something like `broadwell` or `excavator`.  See [these docs](https://spack.readthedocs.io/en/latest/packaging_guide.html#querying-the-target-microarchitecture).

Reply:
I definitely missed this pattern in 41f8f097e94ef8ac615fe53a0d01ee0629009204. @dorier Since #3206 the preferred way to do these checks is:
```python
if spec.target.family == 'x86_64':
    ....
```
Also note that with the naming convention we use `x86` denotes 32-bit, so it might be likely that above you need to change it to  `x86_64`.

Reply:
Using the `in` keyword with a `Target` object has a different meaning, see review below.